validator/client/src/main/java/tech/pegasys/teku/validator/client/BeaconProposerPreparer.java

```diff
@@ -141,6 +147,24 @@ public Optional<Eth1Address> getFeeRecipient(final BLSPublicKey publicKey) {
         .or(() -> defaultFeeRecipient);
   }
 
+  @Override
+  public Optional<UInt64> getGasLimit(BLSPublicKey publicKey) {
+    if (validatorIndexCannotBeResolved(publicKey)) {
+      return Optional.empty();
+    }
+    return maybeProposerConfig
+        .flatMap(config -> getGasLimitFromProposerConfig(config, publicKey))
+        .or(() -> runtimeProposerConfig.getGasLimitForPubKey(publicKey))
+        .or(
+            () ->
+                maybeProposerConfig.flatMap(
+                    proposerConfigProvider ->
+                        proposerConfigProvider.getDefaultConfig().getBuilder().isPresent()
+                            ? proposerConfigProvider.getDefaultConfig().getGasLimit()
+                            : Optional.empty()))
+        .or(() -> Optional.ofNullable(defaultGasLimit));
```

Comment:
this should likely be a lot closer to the fee recipient also, flatmap should be handling empty so all the isPresent etc can disappear